WordPress/gutenberg
# Data views list layout: Re-organise actions display

## What?
This PR attempts to address two issues in list layout.

The first is the awkward alignment of the title when all fields are toggled off:

<img width="378" alt="Screenshot 2024-07-01 at 16 37 41" src="https://github.com/WordPress/gutenberg/assets/846565/7e83956c-00d3-4e71-a231-37140db405a3">

The second is the reflow that occurs when hovering/selecting a row that contains a field with a very long value, or just many fields:


https://github.com/WordPress/gutenberg/assets/846565/da6f4cf0-0aab-4044-b237-32d2a19b3b2a


## How?
Currently the title + fields are in one column, and the actions are in another. When the actions column appears, the title + fields column gets squished to accommodate it causing wrapping.

In this PR I've re-organised the markup so that the title and actions are in a row, with fields beneath. When the actions are revealed the title is squished, but truncated to avoid reflow.

To accomplish this I had to increase the height of the title slightly to align with the action icons. This also increased the height of the title + fields container in a two-row configuration by 4px, so the thumbnail was increased to match. This structure is better described in an image:


<img width="414" alt="guides" src="https://github.com/WordPress/gutenberg/assets/846565/872a03a1-3c5b-4f00-ad53-c08a927edb69">


The result is that regardless of how many fields are visible (or how long the field values are), alignment is maintained:

| Title + 1 row of fields | Title + several rows of fields | Title alone |
| --- | --- | --- |
| <img width="414" alt="Screenshot 2024-07-02 at 10 39 26" src="https://github.com/WordPress/gutenberg/assets/846565/f12a104a-449d-4cbf-abba-56e15e2e5bed"> | <img width="398" alt="Screenshot 2024-07-02 at 10 44 09" src="https://github.com/WordPress/gutenberg/assets/846565/10d8ecb8-8463-4bf3-96e1-ff7813436029"> | <img width="406" alt="Screenshot 2024-07-02 at 10 45 02" src="https://github.com/WordPress/gutenberg/assets/846565/be5e3590-2966-4769-8257-ec6deb041bad"> |




## Testing Instructions
1. Navigate to a dataview like pages and switch to list layout.
2. Toggle fields on / off and ensure alignment is maintained, and that the two issues are fixed.
3. Ensure that keyboard navigation works as expected.
